core/connection/manager.go

```diff
@@ -506,16 +490,16 @@ func (manager *connectionManager) waitForConnectedState(stateChannel <-chan Stat
 }
 
 func (manager *connectionManager) consumeConnectionStates(stateChannel <-chan State) {
+	manager.connectionStopped = make(chan struct{})
 	for state := range stateChannel {
 		manager.onStateChanged(state)
 	}
+	close(manager.connectionStopped)
```

Comment:
Not sure if this extra `connectionStopped` is needed. But if it's needed better name would be `connectionStatesDone` etc. Also try to init channels in ctor when possible.

Reply:
Improved